Add an how-to section to disable Ivy descriptor publication with SDK 6. (#1002)
MicroEJ/docs: SDK6UserGuide/howtos/disableIvyDescriptorPublication.rst

.. _sdk_6_disable_ivy_descriptor_publication:

How To Disable Ivy Descriptor Publication
=========================================

> By default, SDK 6 adds an Ivy descriptor file to published artifacts in order for SDK 5 projects to be able to consume the module (e.g. an application built with SDK 5 depending on a library published with SDK 6).

Reviewer: Should we also mention that for this compatibility to work, the module must be published with a specific version pattern?

Author: Actually, there is a page about versioning that explains this issue: https://docs.microej.com/en/latest/SDK6UserGuide/manageVersion.html I've added a link to this page.

Disable Ivy Descriptor Publication by Default
---------------------------------------------

If you don't need SDK 5 backward compatibility, you can disable Ivy descriptor publication by setting the Gradle property ``enable.ivy.descriptor.default=false``.

This property can be set in a ``gradle.properties`` file in ``<user.home>/.gradle/`` to apply it globally, or at the root of the Gradle project to
apply it at project level.

> It can also passed to command line with ``-Penable.ivy.descriptor.default=false`` option.

Reviewer 2: ```suggestion
It can also be passed to command line with ``-Penable.ivy.descriptor.default=false`` option.

```

.. _sdk_6_disable_ivy_descriptor_publication:

How To Disable Ivy Descriptor Publication
=========================================

> By default, SDK 6 publishes an Ivy descriptor file in order to allow SDK 5 projects to fetch the published artifacts.

Reviewer: I am not sure but I would suggest:
```suggestion
By default, SDK 6 adds an Ivy descriptor file to published artifacts in order for SDK 5 projects to be able to consume the module (e.g. an SDK 5 application built with SDK 5 depending on a library published with SDK 6).
```

Disable Ivy Descriptor Publication by Default
---------------------------------------------

If you don't need SDK 5 backward compatibility, you can disable Ivy descriptor publication by setting the Gradle property ``enable.ivy.descriptor.default=false``.

This property can be set in a ``gradle.properties`` file in ``<user.home>/.gradle/`` to apply it globally, or at the root of the Gradle project to
apply it at project level.
It can also passed to command line with ``-Penable.ivy.descriptor.default=false`` option.

.. note::

    You can get more information about Gradle properties in `Gradle offical documentation <https://docs.gradle.org/current/userguide/build_environment.html#sec:gradle_configuration_properties>`__.
  

Force Ivy Descriptor Publication When Disabled by Default
---------------------------------------------------------

> Even though Ivy descriptor publication is globally disabled with the Gradle property, it can be forced in a project by adding ``withIvyDescriptor()`` to the ``microej`` block

Reviewer: ```suggestion
Even when Ivy descriptor publication is globally disabled with the Gradle property, it can be forced in a project by adding ``withIvyDescriptor()`` to the ``microej`` block

```

.. _sdk_6_disable_ivy_descriptor_publication:

How To Disable Ivy Descriptor Publication
=========================================

> By default, SDK 6 adds an Ivy descriptor file to published artifacts in order for SDK 5 projects to be able to consume the module (e.g. an SDK 5 application built with SDK 5 depending on a library published with SDK 6).

Reviewer 2: ```suggestion
By default, SDK 6 adds an Ivy descriptor file to published artifacts in order for SDK 5 projects to be able to consume the module (e.g. an application built with SDK 5 depending on a library published with SDK 6).

```

Disable Ivy Descriptor Publication by Default
---------------------------------------------

If you don't need SDK 5 backward compatibility, you can disable Ivy descriptor publication by setting the Gradle property ``enable.ivy.descriptor.default=false``.

This property can be set in a ``gradle.properties`` file in ``<user.home>/.gradle/`` to apply it globally, or at the root of the Gradle project to
apply it at project level.
It can also passed to command line with ``-Penable.ivy.descriptor.default=false`` option.

.. note::

>     You can get more information about Gradle properties in `Gradle offical documentation <https://docs.gradle.org/current/userguide/build_environment.html#sec:gradle_configuration_properties>`__.

Reviewer 2: We have an HowTo page about System Properties, so we can refer it here: https://docs.microej.com/en/feature-sdk6_disable_ivy_publication/SDK6UserGuide/howtos/gradleSystemProperty.html

Author: I've removed the note because it is not necessary. I didn't redirect to the page you gave because it is not a system property.

  

Force Ivy Descriptor Publication When Disabled by Default
---------------------------------------------------------

Even when Ivy descriptor publication is globally disabled with the Gradle property, it can be forced in a project by adding ``withIvyDescriptor()`` to the ``microej`` block

> in ``build.gradle.kts`` file::

Reviewer 2: Why is this part of the sentence on a new line instead of after `to the ``microej`` block`?

Author: fixed
